backend/curriculum_tracking/helpers.py

```diff
@@ -182,3 +187,44 @@ def agile_card_reviews_outstanding(user):
 def pull_request_reviews_outstanding(user):
     # TODO: only implement this once the github webhook is healthier
     return []
+
+
+def get_formatted_duration_string(duration):
+    if duration:
+        duration = duration
+        seconds = duration.total_seconds()
+        days, remainder = divmod(seconds, 86400)
+        hours, remainder = divmod(remainder, 3600)
+        minutes, remainder = divmod(remainder, 60)
+
+        return f"{int(days)} days, {int(hours)} hours, {int(minutes)} minutes"
+
+    return None
+
+
+def get_blant_duration(id, card_status):
+    card_started_logs = sorted(
+        LogEntry.objects.filter(
+            Q(event_type__name=CARD_STARTED),
+            object_1_id=id,
+        ),
+        key=lambda log: log.timestamp,
+    )
+
+    card_completed_logs = sorted(
+        LogEntry.objects.filter(
+            Q(event_type__name=CARD_MOVED_TO_COMPLETE),
+            object_1_id=id,
+        ),
+        key=lambda log: log.timestamp,
+    )
+
+    if (
+        card_status != models.AgileCard.COMPLETE
+        or not card_started_logs
+        or not card_completed_logs
+    ):
+        return None
+
+    duration = card_completed_logs[-1].timestamp - card_started_logs[-1].timestamp
```

Comment:
I think this duration should be kept as a timestamp. We can format it into whatever string we want when we need to, maybe in the frontend or wherever this info will be used.

Reply:
I fully agree with Ngoako.

Consider how the data will be used. It should only be formatted for display when it is displayed.

If we wanted to check what is the average duration for a specific project we would need to jump through some crazy hoops here.